app/src/server/tasks/evaluateTestSetEntries.task.ts

```diff
@@ -540,8 +538,10 @@ export const queueEvalJobsForEval = async (datasetEvalId: string) => {
   for (const datasetEvalDatasetEntry of datasetEval.datasetEvalDatasetEntries) {
     for (let i = 0; i < datasetEval.outputSources.length; i++) {
       for (let j = i + 1; j < datasetEval.outputSources.length; j++) {
-        const firstOutputSource = datasetEval.outputSources[i];
-        const secondOutputSource = datasetEval.outputSources[j];
+        const [firstOutputSource, secondOutputSource] = shuffle([
```

Comment:
I would shuffle the order of output sources within the task itself. We're also queueing jobs in `queueHeadToHeadEvalJobsForTestingEntry`, so we would need to shuffle the order there as well if we do it before calling the task.

Reply:
Good call, didn't realize that was being enqueued in two places.

I ended up refactoring to call the same enqueue method from both places since that'll make this kind of slip-up less likely in the future. I ended up doing the shuffling there, although in retrospect doing it within the job like you suggested probably would have worked at least as well.